src/componentManifest.ts

```diff
@@ -143,18 +160,59 @@ export function makeCompanionManifest({
   hasSettings: boolean;
   buildId: string;
 }) {
-  const manifest: CompanionManifest = {
-    manifestVersion: 2,
-    companion: { main: componentTargets.companion.output },
-    ...makeCommonManifest({
-      projectConfig,
-      buildId,
-      apiVersion: apiVersions(projectConfig).companionApi,
-    }),
-  };
+  let companionEntryPoint: string;
+  let settingsEntryPoint: string;
+
+  return new Transform({
+    objectMode: true,
 
-  if (hasSettings) {
-    manifest.settings = { main: componentTargets.settings.output };
-  }
-  return gulpFile(manifestPath, JSON.stringify(manifest));
+    transform(file: Vinyl, _, next) {
+      const componentType = file.componentType;
+
+      const isEntryPoint: boolean | undefined = file.isEntryPoint;
+      if (isEntryPoint && componentType === ComponentType.COMPANION) {
+        companionEntryPoint = file.relative;
+      }
+      if (isEntryPoint && componentType === ComponentType.SETTINGS) {
+        settingsEntryPoint = file.relative;
+      }
```

Comment:
I wonder if it should be an error for a file that is an entry point for neither companion nor settings to pass through this plugin.